### DISCUSSION TITLE
Plots in CML/CI/automated workflows

### DISCUSSION THREAD

**Discussion by dberenbaum** [treeverse/dvc#7183] [General]
What is the best way to make plots outputs available in CML/CI/automated workflows?

There is already `--show-vega`, which CML uses in its example: https://cml.dev/doc/cml-with-dvc. However, this requires a specific target and doesn't work for image files.

`--json` has been added to support other products with both vega and paths to image files. However, this is harder to use in cases like the CML example above where the need is to produce a plot in markdown from a vega plot. 

See https://github.com/iterative/dvc/issues/7093#issuecomment-997887278 for more background.

**Comment by pared** [comment]
I believe we require 2 flags: 
- one (currently `json`) has to evolve into JSON returning data structure similar to what `dvc.plots.collect` returns, in order to support vscode requirements, mainly data separated from the template and error handling (#7692)
- second, that would return already filled templates, for use cases where user needs only filled templates, currently, to some extent `--show-vega` satisfies this requirement but we need it to return all plots for a particular command, probably map `plot_id/path -> filled template`.

**Reply by dberenbaum** [reply]
What should each return for image files?

**Reply by pared** [reply]
`--json` will return path to image saved in workspace. `--show-vega` will not do anything.
We can leave it as is, or we can allow also for returning of base64 encoded images.
